src/core/geometry/junction/Junction.cpp

```diff
@@ -0,0 +1,47 @@
+#include "Junction.h"
+#include <stdexcept>
+
+namespace semba::geometry::junction {
+	const std::string Junction::INVALID_ATTRIBUTE_MESSAGE = "Error: There must be at least 2 coordinates to unite!";
+
+
+	Junction::Junction(std::string & name)
+	{
+		this->name = name;
+	}
+
+	std::string const& Junction::getName() const
+	{
+		return name;
+	}
+
+	std::string Junction::getName()
+	{
+		return name;
+	}
+
+	std::vector<std::vector<CoordIdForJunctions> > Junction::getUnitedCoordIds() const
+	{
+		return unitedCoordIds;
+	}
+
+	void Junction::addCoordinatesToUnite(std::initializer_list<CoordIdForJunctions> coordinates)
```

Comment:
I like this initializer_list thing. However, is there any reason to pass by value here? In the next method you are passing by const reference, which is usually preferred.

Reply:
Addressed in commit e0876b681447e7be6b8220111352cf1fac6c7c78.